presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -201,6 +204,9 @@ public static Slice rescale(Slice decimal, int rescaleFactor)
         }
     }
 
+    /*
+     * Rescale allows values exceeding 10^38 that fits in 127 bits
```

Comment:
I don't think comment notes are enough. At least there should be a different name suggesting that there is 10^38 check or not.